powershell/upload/UploadFile.ps1

```diff
@@ -0,0 +1,533 @@
+# This example shows how to upload a file.
+# Authors: Visma - Transporters Team
+
+[CmdletBinding()]
+Param(
+    [Alias("ConfigPath")]
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = 'Full filePath of the configuration (e.g. C:\Visma\File API\Ftaas.Examples\powershell\upload\config.xml). Default value: set in the code.'
+    )]
+    [string] $_configPath,
+
+    [Alias("RenewCredentials")]
+    [Parameter(
+        Mandatory = $false,
+        HelpMessage = 'Boolean. $true if you want to renew your credentials. $false otherwise'
+    )]
+    [bool] $_renewCredentials = $false
+)
+
+$ErrorActionPreference = "Stop"
+
+# The default value of this parameter is set here because $PSScriptRoot is empty if used directly in Param() through PowerShell ISE.
+if (-not $_configPath) {
+    $_configPath = "$($PSScriptRoot)\config.xml"
+}
+
+Write-Host "========================================================="
+Write-Host "File API example: Upload a file."
+Write-Host "========================================================="
+
+Write-Host "(you can stop the script at any moment by pressing the buttons 'CTRL'+'C')"
+
+#region Configuration
+
+[ConfigurationManager] $configurationManager = [ConfigurationManager]::new()
+
+try {
+    $config = $configurationManager.Get($_configPath)
+}
+catch {
+    [Helper]::EndProgramWithError($_, "Failure retrieving the configuration. Tip: see the README.MD to check the format of the parameters.")
+}
+
+#endregion Configuration
+
+#region Retrieve/Create credentials
+
+$credentialsManager = [CredentialsManager]::new($config.Credentials.Path)
+$credentialsService = [CredentialsService]::new($credentialsManager)
+
+try {
+    if ($_renewCredentials) {
+        $credentials = $credentialsService.CreateNew()
+    }
+    else {
+        $credentials = $credentialsService.Retrieve()
+    }
+}
+catch {
+    [Helper]::EndProgramWithError($_, "Failure retrieving the credentials.")
+}
+
+#endregion Retrieve/Create credentials
+
+#region Retrieve authentication token
+
+$authenticationApiClient = [AuthenticationApiClient]::new($config.Services.AuthenticationTokenApiBaseUrl)
+$authenticationApiService = [AuthenticationApiService]::new($authenticationApiClient)
+
+try {
+    $token = $authenticationApiService.NewToken($credentials.ClientId, $credentials.ClientSecret)
+}
+catch {
+    [Helper]::EndProgramWithError($_, "Failure retrieving the authentication token.")
+}
+
+#endregion Retrieve authentication token
+
+$fileApiClient = [FileApiClient]::new($config.Services.FileApiBaseUrl, $token)
+$fileApiService = [FileApiService]::new($fileApiClient, $config.Upload.TenantId, $config.Upload.BusinessTypeId)
+
+#region Upload file
+
+try {
+    $createdFilePath = $fileApiService.CreateFileToUpload($config.Upload.Path) 
```

Comment:
I prefer to hide the creation of the 'boundary file' inside the UploadFile method.
I don't care too much, though.